### PR TITLE
core[patch]: Move async generator consumption code into local storage context

### PR DESCRIPTION
**Pull request by jacoblee93**
@nfcampos can you have a look please?

### CODE REVIEW

**Comment by jacoblee93** [comment]
CC @hinthornw as well

File: langchain-core/src/utils/stream.ts [hunk 1]
```diff
@@ -208,7 +220,12 @@ export class AsyncGeneratorWithSetup<
       return this.firstResult;
     }
 
-    return this.generator.next(...args);
+    const storage = AsyncLocalStorageProviderSingleton.getInstance();
+    return new Promise((resolve) => {
```
Comment (nfcampos) [inline langchain-core/src/utils/stream.ts:224]: storage.run returns whatever the function returns so no need for the extra promise

File: langchain-core/src/utils/stream.ts [hunk 2]
```diff
@@ -182,23 +184,33 @@ export class AsyncGeneratorWithSetup<
...
     // to each generator is available.
     this.setup = new Promise((resolve, reject) => {
-      this.firstResult = generator.next();
-      if (startSetup) {
-        this.firstResult.then(startSetup).then(resolve, reject);
-      } else {
-        this.firstResult.then((_result) => resolve(undefined as S), reject);
-      }
+      const storage = AsyncLocalStorageProviderSingleton.getInstance();
+      storage.run(params.config, async () => {
```
[the remark this answers is not in the document]
Comment (jacoblee93) [reply langchain-core/src/utils/stream.ts:206]: I'm "resolving" this comment!

**Comment by jacoblee93** [comment]
Updated and added test

**Pull request merged**